test/libsolidity/SolidityNameAndTypeResolution.cpp

```diff
@@ -6536,6 +6536,28 @@ BOOST_AUTO_TEST_CASE(constructor_without_implementation)
 	CHECK_ERROR(text, TypeError, "Constructor must be implemented if declared.");
 }
 
+BOOST_AUTO_TEST_CASE(library_function_without_implementation)
+{
+	char const* text = R"(
+		library L {
+			function f();
```

Comment:
I can see a point of keeping this, since at link time an address can be supplied which has it implemented. Also disallowing this would be a breaking change.